.github/workflows/codeql-analysis.yml

```diff
@@ -37,6 +33,12 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
 
+      - name: Check for file changes
+        id: check
+        uses: ./.github/actions/change-detector/
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Is there a use case where you might use different github tokens for different actions? If not, you could DRY it up further and _not_ pass a token from every action to the change detector, but rather use the secret directly there.

Reply:
this seemed like a limitation / requirement give how GHA works, I was able to make everything else DRY, but not this. We really just need read access to a public repo here (looking up either the PR's file touched, or the diff since last commit for `push` events), so maybe there's a way this can work without passing a token explicitely. Let me test whether it works without passing it.

Reply:
Tested and seems it's just the way reusable actions work - they don't have access to secrets for security reason. The previous shell script had to do something similar, required also the PR_NUMBER to be passed (now I pick it up from env var), and I think it didn't support `push` events properly. So still a step forward.